cmd/earthly/main.go

```diff
@@ -411,10 +420,60 @@ func newEarthlyApp(ctx context.Context, console conslogging.ConsoleLogger) *eart
 		},
 		&cli.StringFlag{
 			Name:        "buildkit-host",
+			Value:       "tcp://127.0.0.1:8372",
 			EnvVars:     []string{"EARTHLY_BUILDKIT_HOST"},
 			Usage:       wrap("The URL to use for connecting to a buildkit host. ", "If empty, earthly will attempt to start a buildkitd instance via docker run"),
 			Destination: &app.buildkitHost,
 		},
+		&cli.IntFlag{
+			Name:        "buildkit-port",
+			Value:       config.DefaultBuildkitPort,
+			EnvVars:     []string{"EARTHLY_BUILDKIT_PORT"},
+			Usage:       wrap("The URL to use for connecting to a buildkit host. ", "If empty, earthly will attempt to start a buildkitd instance via docker run"),
+			Destination: &app.buildkitPort,
```

Comment:
Could this port just be inferred from `--buildkit-host`?

Reply:
Actually, I can see why these are separate settings - the port is the buildkit listening port and the URL is what Earthly uses to connect to it. It's confusing at first, but I can't think of a way to improve this.

Reply:
Maybe call it "local port" rather than just "port"? `--buildkit-local-port` and `EARTHLY_BUILDKIT_LOCAL_PORT`

Reply:
Right now, we do. But I also added a deprecation notice if you include the port here. The remote settings are now three parts:

* Root host
* Buildkit port (uses root host)
* Debugger port (uses root host)

I did not want to allow the hosts to diverge between these two parts.

Reply:
Local port would be a misnomer; since it could be used to specify remote ports too (assuming the root URL was not considered local"). I do agree that naming could be improved though.